src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NetworkPeerBehavior.cs

```diff
@@ -68,14 +62,15 @@ public void Detach()
                     return;
 
                 this.DetachCore();
-                foreach (IDisposable dispo in this.disposables)
-                    dispo.Dispose();
-
-                this.disposables.Clear();
-                this.AttachedPeer = null;
             }
         }
 
+        ///  <inheritdoc />
+        public virtual void Dispose()
+        {
+            this.AttachedPeer = null;
```

Comment:
Whats the point? so it GCed?

Reply:
So behavior throws when we are using it after it's disposed. Easier to catch such design bugs that way.